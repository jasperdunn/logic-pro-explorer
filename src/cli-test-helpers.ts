import { execSync } from 'child_process'
import { resolve } from 'path'

export function runCommand(command: string): string {
  return execSync(`node ${resolve(__dirname, '../dist/src/index.js')} ${command}`).toString()
}
