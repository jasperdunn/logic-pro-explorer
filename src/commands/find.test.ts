import { runCommand } from '../cli-test-helpers'

describe('find command', () => {
  test('should display the help message', () => {
    expect(runCommand('find -h')).toMatchSnapshot()
  })
})
