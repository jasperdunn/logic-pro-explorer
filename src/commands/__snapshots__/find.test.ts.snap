// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`find command > should display the help message 1`] = `
"Usage: logic-pro-explorer find [options]

Find projects that are using the selected components.

Options:
  -h, --help  display help for command
"
`;
